documentation/development-docs/patterns-for-standard-cmdlets.md

```diff
@@ -0,0 +1,996 @@
+                var result = new PSFoo(this.MySDKClient.Foo.Update(this.ResourceGroupName, this.Name, foo)
+                WriteObject(result);
+            }
+        }
+    }
+}
+```
+
+</p>
+</details>
+
+## Child Resources
+
+### `Get-*` cmdlet
+
+All child resources should have a `Get-*` cmdlet that allows users to list the child resources under a parent resource, as well as get a specific resource. In addition, users should be able to provide the resource id of the resource they want to get, and the cmdlet will parse the string to get the necessary identity information.
+
+#### Parameter sets
+
+To enable the scenarios mentioned previously, the cmdlet will need three parameter sets:
+
+```
+Get-AzureRmChildFoo -ResourceGroupName <String> -ParentName <String> [-Name <String>]
+
+Get-AzureRmChildFoo -FooObject <PSFoo> [-Name <String>]
+
+Get-AzureRmChildFoo -ResourceId <String>
+```
+
+The first parameter set has mandatory `-ResourceGroupName` and `-ParentName` parameters to get the identity information about the parent resource, and then an optional `-Name` parameter to allow the user to either list all child resources contained in the given parent or get the specific child resource. The second parameter set has required `-FooObject` parameter, which can be piped from the parent `Get-*` or `Set/Update-*` cmdlet, and an optional `-Name` parameter. The third parameter set has a required `-ResourceId` parameter, which allows the user to get a specific child resource by resource id.
+
+#### C# example
+
+<details><summary>Click to expand example</summary>
+<p>
+
+```cs
+namespace Microsoft.Azure.Commands.Foo
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmChildFoo", DefaultParameterSetName = "GetByFooNameParameterSet"), OutputType(typeof(PSChildFoo))]
+    public class GetChildFooCommand : FooBaseCmdlet
+    {
+        [Parameter(Mandatory = true, ParameterSetName = "GetByFooNameParameterSet")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = "GetByFooNameParameterSet")]
+        [ValidateNotNullOrEmpty]
+        public string ParentName { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = "GetByFooNameParameterSet")]
+        [Parameter(Mandatory = false, ParameterSetName = "GetByFooObjectParameterSet")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = "GetByFooObjectParameterSet")]
+        [ValidateNotNull]
+        public PSFoo FooObject { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = "Get ByResourceIdParameterSet")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.IsParameterBound(c => c.ResourceId))
```

Comment:
why not use [parameter set](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Billing/Commands.Billing/Cmdlets/BillingPeriods/GetAzureRmBillingPeriod.cs#L39) here?

Reply:
Same response as above.